ff/serving/metadata-rest/metadata-rest-server.go

```diff
@@ -0,0 +1,611 @@
+}
+
+type LabelVariantResource struct {
+	Created      time.Time              `json:"created"`
+	Description  string                 `json:"description"`
+	Entity       string                 `json:"entity"`
+	Name         string                 `json:"name"`
+	Owner        string                 `json:"owner"`
+	Provider     string                 `json:"provider"`
+	Type         string                 `json:"type"`
+	Variant      string                 `json:"variant"`
+	Source       metadata.NameVariant   `json:"source"`
+	TrainingSets []metadata.NameVariant `json:"trainingsets"`
+}
+
+type LabelResource struct {
+	AllVariants    []string                        `json:"all-versions"`
+	DefaultVariant string                          `json:"default-variant"`
+	Name           string                          `json:"name"`
+	Variants       map[string]LabelVariantResource `json:"versions"`
+}
+
+type EntityResource struct {
+	Name         string                 `json:"name"`
+	Description  string                 `json:"description"`
+	Features     []metadata.NameVariant `json:"features"`
+	Labels       []metadata.NameVariant `json:"labels"`
+	TrainingSets []metadata.NameVariant `json:"trainingsets"`
+}
+
+type UserResource struct {
+	Name         string                 `json:"name"`
+	Features     []metadata.NameVariant `json:"features"`
+	Labels       []metadata.NameVariant `json:"labels"`
+	TrainingSets []metadata.NameVariant `json:"trainingsets"`
+	Sources      []metadata.NameVariant `json:"sources"`
+}
+
+type ModelResource struct {
+	Name         string                 `json:"name"`
+	Description  string                 `json:"description"`
+	Features     []metadata.NameVariant `json:"features"`
+	Labels       []metadata.NameVariant `json:"labels"`
+	TrainingSets []metadata.NameVariant `json:"trainingsets"`
+}
+
+type ProviderResource struct {
+	Name         string                 `json:"name"`
+	Description  string                 `json:"description"`
+	Type         string                 `json:"type"`
+	Software     string                 `json:"software"`
+	Team         string                 `json:"team"`
+	Sources      []metadata.NameVariant `json:"sources"`
+	Features     []metadata.NameVariant `json:"features"`
+	Labels       []metadata.NameVariant `json:"labels"`
+	TrainingSets []metadata.NameVariant `json:"trainingsets"`
+}
+
+type FetchError struct {
+	StatusCode int
+	Type       string
+}
+
+func (m *FetchError) Error() string {
+	return fmt.Sprintf("Error %d: Failed to fetch %s", m.StatusCode, m.Type)
+}
+
+func (m MetadataServer) readFromFeature(feature *metadata.Feature) (map[string]FeatureVariantResource, *FetchError) {
+	variantMap := make(map[string]FeatureVariantResource)
+	variants, err := feature.FetchVariants(m.client, context.Background())
+	if err != nil {
+		fetchError := &FetchError{StatusCode: 500, Type: "feature variants"}
+		m.logger.Errorw("Failed to fetch variants", "Error", err)
```

Comment:
Use:
fetchError.Error() for the message

Reply:
Currently the code doesn't print the fetch error here, but rather logs the fetcherror in the method this function returns its error to.

Reply:
Why not do it like this:
		m.logger.Errorw(fetchError.Error(), "Internal Error", err)
Seems better to me, but if you'd prefer to do it this way, I don't have a super strong opinion